client/src/components/Calendar.jsx

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import moment from 'moment';
+import styled from 'styled-components';
+
+
+class Calendar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      dateObject: moment(),
+      leftMonth: moment(),
+      rightMonth: moment().add(1, 'months'),
+      leftDays: null,
+      rightDays: null,
+      dateSelected: null
+    };
+  }
+
+  previousMonth() {
+    let left = moment(this.state.leftMonth).subtract(1, 'months');
+    let right = moment(this.state.rightMonth).subtract(1, 'months');
+    this.setState({
+      leftMonth: left,
+      rightMonth: right
+    });
+  }
+    
+  nextMonth() {
+    let left = moment(this.state.leftMonth).add(1, 'months');
+    let right = moment(this.state.rightMonth).add(1, 'months');
+    this.setState({
+      leftMonth: left,
+      rightMonth: right
+    });
+  }
+
+  firstDayOfMonth(leftOrRight) {
+    let firstDay = moment(leftOrRight).startOf('month').format('d');
+    return firstDay;
+  }
+
+  render() {
+    const eachDayOfWeek = moment.weekdaysShort().map(day => (
+      <DayOfMonth key={day}>
+        {day.slice(0, -1)}
```

Comment:
it's hard for me to grasp what you are doing here.  Do you mind adding some comment or changing the variable names?

Given "slice" is usually called on an Array/String, I'm having hard time guessing what is stored in a variable called `day` as an `Array/String`.

Reply:
Thank you for pointing this out.